packages/skygear-react-native/android/src/main/java/io/skygear/reactnative/SGSkygearReactNativeModule.java

```diff
@@ -117,6 +131,65 @@ public void getCredentialStateForUserID(String userID, Promise promise) {
         promise.reject("EUNSPECIFIED", "Sign in with Apple is not supported on Android. Please use loginOAuthProvider or linkOAuthProvider instead.");
     }
 
+    @ReactMethod
+    public void getAnonymousKey(String kid, Promise promise) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            promise.reject("EUNSPECIFIED", "Anonymous authentication is not supported on Android before version M.");
+            return;
+        }
+
+        try {
+            if (kid == null) {
+                kid = UUID.randomUUID().toString();
+            }
+            Map<String, Object> header = new HashMap<>();
```

Comment:
On android, we need to use WritableMap and WritableArray to write JavaScript object and array respectively. Unlike iOS, native Java map and array types do not work :(

This page contains some examples https://reactnative.dev/docs/native-modules-android